zap/src/main/java/org/zaproxy/zap/spider/parser/SpiderHtmlParser.java

```diff
@@ -39,14 +39,45 @@
  */
 public class SpiderHtmlParser extends SpiderParser {
 
-    /** The Constant URL_PATTERN defining the pattern for a meta url. */
+    /** The Constant URL_PATTERN defining the pattern for a meta URL. */
     static final Pattern URL_PATTERN =
             Pattern.compile("url\\s*=\\s*[\"']?([^;'\"]+)", Pattern.CASE_INSENSITIVE);
 
     private static final Pattern PLAIN_COMMENTS_URL_PATTERN =
             Pattern.compile(
                     "(?:http(?:s?):)?//[^\\x00-\\x1f\"'\\s<>#()\\[\\]{}]+",
                     Pattern.CASE_INSENSITIVE);
+    /**
+     * Functional interface that allows custom handling of URLs retrieved in attributes, to be
+     * manipulated before the URL is sent to a {@link SpiderParser:processURL} call.
+     */
+    @FunctionalInterface
+    private interface CustomUrlProcessor {
+        void process(HttpMessage message, int depth, String localURL, String baseURL);
```

Comment:
Allow to pass the spider tasks and make `SrcSetProcessor` a class method.

Reply:
Sorry, called it wrong, I meant the spider parser not task, i.e. `void process(SpiderHtmlParser, HttpMessage, int, String, String);`. `SrcSetProcessor` can then be a class method and use a method reference in the `processAttributeElement` call, instead of creating a new instance each time.

Reply:
There is actually no need to add the SpiderHtmlParser, just the method reference is sufficient. 
Changed the code to reflect that

Reply:
Sounds good.